### PR TITLE
return false if it is not wc_product

### LINKED ISSUE #4255: Delay JS (Error 500):  Uncaught Error: Call to a member function get_gallery_image_ids() on bool
**Before submitting an issue please check that you’ve completed the following steps:**
- Made sure you’re on the latest version ✅ - 3.9.1.1
- Used the search feature to ensure that the bug hasn’t been reported before ✅

**Describe the bug**
We are using the function wc_get_product() and that function can return these possible values: WC_Product | null | false ([Reference](https://woocommerce.github.io/code-reference/namespaces/default.html#function_wc_get_product))

And here it seems that we are not handling all the possible return values: 

https://github.com/wp-media/wp-rocket/blob/a38f5b6aa1853d221b97ebbaa9ef9638cafd7e2c/inc/ThirdParty/Plugins/Ecommerce/WooCommerceSubscriber.php#L515

That causes when the function returns false, we get and error when trying this:

https://github.com/wp-media/wp-rocket/blob/a38f5b6aa1853d221b97ebbaa9ef9638cafd7e2c/inc/ThirdParty/Plugins/Ecommerce/WooCommerceSubscriber.php#L516

The full error is:

> PHP Fatal error:  Uncaught Error: Call to a member function get_gallery_image_ids() on bool in /srv/users/signetprelive/apps/signetprelive/public/wp-content/plugins/wp-rocket/inc/ThirdParty/Plugins/Ecommerce/WooCommerceSubscriber.php:516

**Expected behavior**
The site shouldn't throw an error 500.

**Backlog Grooming (for WP Media dev team use only)**
- [ ] Reproduce the problem
- [ ] Identify the root cause
- [ ] Scope a solution
- [ ] Estimate the effort

### PR DESCRIPTION
**Pull request by mostafa-hisham**
## Description

return false if it is not wc_product

Fixes #4255 

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

## Is the solution different from the one proposed during the grooming?
No

## How Has This Been Tested?

- [x] unit and integration tests

# Checklist:

Please delete the options that are not relevant.

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: inc/ThirdParty/Plugins/Ecommerce/WooCommerceSubscriber.php [hunk 1]
```diff
@@ -513,6 +513,9 @@ private function get_nonce_actions() {
 	 */
 	private function product_has_gallery_images() {
 		$product = wc_get_product( get_the_ID() );
+		if ( ! $product || empty( $product ) ) {
```
Comment (remyperona) [inline inc/ThirdParty/Plugins/Ecommerce/WooCommerceSubscriber.php]: empty will check for both null and false, so we don't need the first part of the condition

**Review by remyperona — CHANGES_REQUESTED** [review]

**Review by remyperona — APPROVED** [review]

**Review by piotrbak — APPROVED** [review]
Good to go!
[testrail-report-249.pdf](https://github.com/wp-media/wp-rocket/files/7117334/testrail-report-249.pdf)

**Comment by katielgc** [comment]
Hi all, could you let me know when this will be merged into the master for deployment on our site?

**Comment by iCaspar** [comment]
@katielgc We expect, pending final QA, to merge and release this week. 🤞

**Pull request merged**